<launch>
     <arg name="gui" default="false" />
     <param name="use_gui" value="$(arg gui)"/>
     <param name="rate" value="50.0"/>

     <!-- send the g1 robot XML to param server -->
     <param name="robot_description" textfile="$(find g1_description)/g1_23dof.urdf"/>
        
    <group if="$(arg gui)">
        <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui">
     	    <param name="publish_default_efforts" value="True"/>
        </node>
    </group>
    
    <rosparam param="left_foot_list">[
      "left_foot_upper_right",
      "left_foot_lower_right",
      "left_foot_upper_left",
      "left_foot_lower_left"]
    </rosparam>
    
    <rosparam param="right_foot_list">[
      "right_foot_upper_right",
      "right_foot_lower_right",
      "right_foot_upper_left",
      "right_foot_lower_left"]
    </rosparam>

    <rosparam param="q_init">
    [
    0.0,
    0.0,
    0.0,  
    0.0,
    0.0,
    0.0,
    1.0,  
    -0.6, 
    0.0, 
    0.0,
    1.2, 
    -0.6, 
    0.0, 
    -0.6,
    0.0,
    0.0,
    1.2,
    -0.6,
    0.0,
    0.0,
    0.0,
    0.0,
    0.0,
    0.0,
    0.0,
    0.0,
    0.0,
    0.0,
    0.0,
    0.0 
    ]
    </rosparam>

    <!-- start robot state publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen" >
        <param name="publish_frequency" type="double" value="250.0" />
    </node> 

</launch>
